src/Param.cc

```diff
@@ -304,140 +314,149 @@ bool Param::ValueFromString(const std::string &_value)
       numericBase = 16;
     }
 
-    if (this->dataPtr->typeName == "bool")
+    if (_typeName == "bool")
     {
       if (lowerTmp == "true" || lowerTmp == "1")
       {
-        this->dataPtr->value = true;
+        _valueToSet = true;
       }
       else if (lowerTmp == "false" || lowerTmp == "0")
       {
-        this->dataPtr->value = false;
+        _valueToSet = false;
       }
       else
       {
         sdferr << "Invalid boolean value\n";
         return false;
       }
     }
-    else if (this->dataPtr->typeName == "char")
+    else if (_typeName == "char")
     {
-      this->dataPtr->value = tmp[0];
+      _valueToSet = tmp[0];
     }
-    else if (this->dataPtr->typeName == "std::string" ||
-             this->dataPtr->typeName == "string")
+    else if (_typeName == "std::string" ||
+             _typeName == "string")
     {
-      this->dataPtr->value = tmp;
+      _valueToSet = tmp;
     }
-    else if (this->dataPtr->typeName == "int")
+    else if (_typeName == "int")
     {
-      this->dataPtr->value = std::stoi(tmp, nullptr, numericBase);
+      _valueToSet = std::stoi(tmp, nullptr, numericBase);
     }
-    else if (this->dataPtr->typeName == "uint64_t")
+    else if (_typeName == "uint64_t")
     {
       StringStreamClassicLocale ss(tmp);
       std::uint64_t u64tmp;
 
       ss >> u64tmp;
-      this->dataPtr->value = u64tmp;
+      _valueToSet = u64tmp;
     }
-    else if (this->dataPtr->typeName == "unsigned int")
+    else if (_typeName == "unsigned int")
     {
-      this->dataPtr->value = static_cast<unsigned int>(
+      _valueToSet = static_cast<unsigned int>(
           std::stoul(tmp, nullptr, numericBase));
     }
-    else if (this->dataPtr->typeName == "double")
+    else if (_typeName == "double")
     {
-      this->dataPtr->value = std::stod(tmp);
+      _valueToSet = std::stod(tmp);
     }
-    else if (this->dataPtr->typeName == "float")
+    else if (_typeName == "float")
     {
-      this->dataPtr->value = std::stof(tmp);
+      _valueToSet = std::stof(tmp);
     }
-    else if (this->dataPtr->typeName == "sdf::Time" ||
-             this->dataPtr->typeName == "time")
+    else if (_typeName == "sdf::Time" ||
+             _typeName == "time")
     {
       StringStreamClassicLocale ss(tmp);
       sdf::Time timetmp;
 
       ss >> timetmp;
-      this->dataPtr->value = timetmp;
+      _valueToSet = timetmp;
+    }
+    else if (_typeName == "ignition::math::Angle" ||
```

Comment:
Glad you caught that this was missing. It's weird that "angle" is never used by any of the description files but it's one of the types in the variant though. Mind creating an issue?

Reply:
How's this? https://github.com/ignitionrobotics/sdformat/issues/654

Reply:
Thanks!